scripts/upgrade_cfme.py

```diff
@@ -0,0 +1,216 @@
+        api_url = "https://{ip}/api/".format(ip=self.address)
+        auth = ("admin", "smartvm")
+        si = requests.get(api_url, verify=False, auth=auth).json()["server_info"]
+        assert si['version'].split("-", 1)[0] == self.dest_version, "The upgrade filed!"
+        logger.info("The upgrade finished successfully")
+
+    def restart_components(self):
+        logger.info("restarting componates after updating components")
+
+        self.exec_commaned("systemctl restart $APPLIANCE_PG_SERVICE")
+
+        self.ssh_client.run_rake_command("db:migrate")
+        self.ssh_client.run_rake_command("evm:automate:reset")
+        self.exec_commaned("systemctl restart rh-postgresql95-postgresql")
+
+
+def main():
+    parser = argparse.ArgumentParser(epilog=__doc__,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("-c", "--config", help="yaml file with configurations",
+                        dest="config", action="store", default=False)
+    args = parser.parse_args()
+
+    params = {}
+    try:
+        logger.info("loading config file")
+        with open(args.config, 'r') as stream:
+            params = yaml.load(stream)
+    except Exception as ex:
+        logger.exception(ex.message)
+
+    # filed name: (error msg, is mandatory, default value)
+    requierd_fildes = {"address": {"msg": "Appliance address is missing",
+                                   "is_mandatory": True, "default": None},
+                       "username": {"msg": "subscription manager user name is missing",
+                                    "is_mandatory": True, "default": None},
+                       "password": {"msg": "subscription manager password is missing",
+                                    "is_mandatory": True, "default": None},
+                       "skip_validation": {"msg": "skip validation key not found on config file, "
+                                           "assuming version validation is required",
+                                           "is_mandatory": False, "default": True},
+                       "dest_version": {"msg": "destination version is missing"
+                       if not params.get("skip_validation", False) else "",
+                                        "is_mandatory": not params.get("skip_validation", False),
+                                        "default": None},
+                       "repo": {"msg": "repolist is missing",
+                                "is_mandatory": True, "default": None}}
+    for key in requierd_fildes:
+        if key not in params:
+            try:
+                if requierd_fildes[key]["is_mandatory"]:
+                    raise RuntimeError(requierd_fildes.get(key).get("msg"))
+            except Exception as ex:
+                logger.exception(ex.message)
+            else:
+                params.update({key: requierd_fildes.get(key).get("default")})
```

Comment:
Isn't this equivalent to
`params[key] = requierd_fildes.get(key).get("default")`
?